international-telephone-input/functions/lookup.js

```diff
@@ -0,0 +1,55 @@
+/**
+ *  Lookup - validate a phone number
+ *
+ *  This function will tell you whether or not a phone number is valid using Twilio's Lookup API
+ *
+ *  Parameters:
+ *  "phone" - string - phone number in E.164 format (https://www.twilio.com/docs/glossary/what-e164)
+ *
+ *  Returns JSON
+ *  {
+ *    "success": boolean,
+ *    "error": string      // not present if success is true
+ *  }
+ */
+
+exports.handler = function (context, event, callback) {
+  const response = new Twilio.Response();
+  response.appendHeader("Content-Type", "application/json");
+
+  // uncomment to support CORS
+  // response.appendHeader('Access-Control-Allow-Origin', '*');
+  // response.appendHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
+  // response.appendHeader('Access-Control-Allow-Headers', 'Content-Type');
+
+  if (typeof event.phone === "undefined") {
+    response.setBody({
+      success: false,
+      error: "Missing parameter; please provide a phone number.",
+    });
+    response.setStatusCode(400);
+    return callback(null, response);
+  }
+
+  const client = context.getTwilioClient();
+
+  client.lookups
+    .phoneNumbers(event.phone)
+    .fetch()
+    .then((resp) => {
+      response.setStatusCode(200);
+      response.setBody({
+        success: true,
+      });
+      callback(null, response);
+    })
+    .catch((error) => {
+      console.log(error);
```

Comment:
It's maybe not important, but if this error is a 404 (20404) then it just means invalid phone number whereas if the error is a 5xx then perhaps something else is going on and we should log it as an error so it triggers the debugger.

I might choose to do something like:

```js
.catch((error) => {
  if (error.code !== 20404) {
    console.error(error);
  } else {
    // assuming we want to log invalid numbers for informational purposes
    console.log(error);
  }
  response.setStatusCode(error.status);
  // ...and so on
```

Reply:
Ah good catch, I'm going to leave this one since it propagates the error message back to the front end so the user would still see the 404 - for a sample app like this I'm ok with a little less code.